packages/repo-tools/src/commands/api-reports/api-reports.ts

```diff
@@ -102,6 +104,14 @@ export const buildApiReports = async (paths: string[] = [], opts: Options) => {
     });
   }
 
+  if (selectedPackageDirs.length > 0) {
```

Comment:
I think we need to run this separately by default, maybe a `--knip` flag, similar to `--docs`. It's a tradeoff, but in my experience it's not super uncommon to need to do a full API reports build in different situations, and this would add quite a lot of extra time to that. Tbh given the other flags that are present it might even make sense to move this to its own command. Still like `--allow-warnings` won't apply here, and we may want to add some of our own knip specific flags. Perhaps a `backstage-repo-tools knip-reports`?

Reply:
Moved this to own command